plugin-lifecycle/src/main/java/com/mapbox/maps/plugin/lifecycle/MapboxLifecyclePluginImpl.kt

```diff
@@ -33,14 +33,23 @@ class MapboxLifecyclePluginImpl : MapboxLifecyclePlugin {
           you need manually invoke the corresponding lifecycle methods in onStart/onStop/onDestroy/onLowMemory methods of the host Activity"""
       )
     } else {
-      val componentCallback = object : ComponentCallbacks {
+      val componentCallback = object : ComponentCallbacks2 {
         override fun onConfigurationChanged(newConfig: Configuration) {
           // no need
         }
 
         override fun onLowMemory() {
           observer.onLowMemory()
         }
+
+        override fun onTrimMemory(level: Int) {
+          when (level) {
+            ComponentCallbacks2.TRIM_MEMORY_RUNNING_CRITICAL, ComponentCallbacks2.TRIM_MEMORY_RUNNING_LOW -> {
+              observer.onLowMemory()
+            }
+            ComponentCallbacks2.TRIM_MEMORY_BACKGROUND, ComponentCallbacks2.TRIM_MEMORY_COMPLETE, ComponentCallbacks2.TRIM_MEMORY_MODERATE, ComponentCallbacks2.TRIM_MEMORY_RUNNING_MODERATE, ComponentCallbacks2.TRIM_MEMORY_UI_HIDDEN -> Unit
```

Comment:
Perhaps log warning here using `Logger.w`?

Reply:
do we need to? I think these will be called pretty frequently, could be noisy to users.

Reply:
yeah, agree